test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/TxnVerbs.java

```diff
@@ -330,6 +353,105 @@ public static HapiContractUpdate contractUpdate(String contract) {
 		return new HapiContractUpdate(contract);
 	}
 
+	/**
+	 * This method enables the developer to upload one or many contract(s) bytecode(s), but without the ability to chain methods
+	 * related to HapiFileCreate.
+	 */
+	public static HapiSpecOperation uploadInitCode(final String... contractsNames) {
+		return withOpContext((spec, ctxLog) -> {
+			List<HapiSpecOperation> ops = new ArrayList<>();
+			for (String contractName : contractsNames) {
+				final var path = getResourcePath(contractName, ".bin");
+				final var file = new HapiFileCreate(contractName);
+				final var updatedFile = updateLargeFile(GENESIS, contractName, extractByteCode(path));
+				ops.add(file);
+				ops.add(updatedFile);
+			}
+			allRunFor(spec, ops);
+		});
+	}
+
+	/**This method enables the developer to upload contract bytecode with the ability to chain methods, related to HapiFileCreate.
+	 The execution of the method requires the op context, therefore the invocation is not straightforward.
+	 Example:
+	 withOpContext(
+	 (spec, noOp) -> {
+	 chainedFileCreate(contract, spec).hasPrecheck(TRANSACTION_OVERSIZE);
+	 chainedFileCreate(contract, spec).contents("").key(KEY_LIST);
+	 }
+	 )
+	 */
+	public static HapiFileCreate uploadInitCode(final String contractName, final HapiApiSpec spec) {
+		final var path = getResourcePath(contractName, ".bin");
+		final var file = new HapiFileCreate(contractName);
+		sourcing(() -> file).execFor(spec);
+		sourcing(() -> updateLargeFile(GENESIS, contractName, extractByteCode(path))).execFor(spec);
+		return file;
+	}
```

Comment:
Do we really need to add complexity to `TxnVerbs` and overload `uploadInitCode` just so we can use it in **one** test case? I think we can delete this method and use `fileUpload` in `ContractCreateSuite.cannotCreateTooLargeContract()`.

Reply:
Solved